core/lib/spree/core/role_configuration.rb

```diff
@@ -0,0 +1,84 @@
+require 'singleton'
+
+module Spree
+  # A class responsible for associating {Spree::Role} with a list of permission sets.
+  #
+  # @see Spree::PermissionSets
+  #
+  # @example Adding order, product, and user display to customer service users.
+  #   Spree::RoleConfiguration.configure do |config|
+  #     config.register_role :customer_service, [
+  #       Spree::PermissionSets::OrderDisplay,
+  #       Spree::PermissionSets::UserDisplay,
+  #       Spree::PermissionSets::ProductDisplay
+  #     ]
+  #   end
+  class RoleConfiguration
+    # An internal structure for the association between a role and a
+    # set of permissions.
+    Role = Struct.new(:name, :permission_sets)
+
+    include Singleton
+    attr_accessor :roles
+
+    # Yields the instance of the singleton, used for configuration
+    # @yield_param instance [Spree::RoleConfiguration]
+    def self.configure
+      yield(instance)
+    end
+
+    # Given an ability, and a user, determine what permissions sets can be activated
+    # on the ability, then activate them.
+    #
+    # This performs can/cannot declarations on the ability, and can modify it's internal permissions
+    #
+    # @param ability [CanCan::Ability] the ability to invoke declarations on
+    # @param user [#spree_roles] the user that holds the spree_roles association.
+    def activate_permissions! ability, user
+      spree_roles = user.spree_roles.pluck(:name)
```

Comment:
This made me realize, maybe it's time to make roles attribute a hash of `{name => Role}`

This would allow something like:

```
def initialize
  @roles = Hash.new do |h, name|
    h[name] = Role.new(name, Set.new)
  end
end
```

This would allow us just iterate over the provided spree_roles, and elimintate the need for `find_role`.